ecommerce/core/templatetags/core_extras.py

```diff
@@ -31,6 +98,90 @@ def settings_value(name):
     return getattr(settings, name)
 
 
+@register.simple_tag(name='is_special_coupon')
+def is_special_coupon(provided_object):
+    """
+    Checks if 'provided_object' contains a coupon which its name meets the format criteria of a special coupon.
+
+    - Requirements: The site.siteconfiguration.custom_settings 'REMOVE_SPECIAL_COUPON_OFFER_PREFIX' key
+    should be set to properly use this tag.
+
+    - Usage: This tag is aimed to be used for conditional logic, as follows.
+        {% load core_extras %}
+
+        {% is_special_coupon order as is_special_coupon %}
+
+        {% if is_special_coupon %}
+                Do something...
+        {% endif %}
+
+    Arguments:
+        provided_object (Order or Basket or OrderDiscount): Order of the purchase,
+        its basket or its OrderDiscount.
+
+    Returns:
+        bool: True if the coupon name is a valid special coupon, False otherwise.
+    """
+    coupon_name = get_coupon_name(provided_object)
+
+    if not coupon_name:
+        return False
+
+    request = get_current_request()
+
+    if not has_request_siteconfiguration(request):
+        return False
+
+    if is_valid_special_coupon(
+            coupon_name,
+            request.site.siteconfiguration.custom_settings.get('REMOVE_SPECIAL_COUPON_OFFER_PREFIX', '')):
```

Comment:
Bad indentation? @Jacatove

Reply:
No, That's not bad indentation. However I know it looks weird. If you agree, we can let a variable which is just used once as how it was before:
```suggestion
    prefix = request.site.siteconfiguration.custom_settings.get('REMOVE_SPECIAL_COUPON_OFFER_PREFIX', '')

    if is_valid_special_coupon(coupon_name, prefix):
        return True
```